tree/treeplayer/src/TDFUtils.cxx

```diff
@@ -114,23 +114,31 @@ std::string TypeID2TypeName(const std::type_info &id)
 
 /// Return a string containing the type of the given branch. Works both with real TTree branches and with temporary
 /// column created by Define. Returns an empty string if type name deduction fails.
+/// Note that for fixed- or variable-sized c-style arrays as well std::vector<T>, the returned type name will be TVec<T>
 std::string
 ColumnName2ColumnTypeName(const std::string &colName, TTree *tree, TCustomColumnBase *tmpBranch, TDataSource *ds)
 {
+   std::string colType;
+
    // if this is a TDataSource column, we just ask the type name to the data-source
-   if (ds && ds->HasColumn(colName)) {
-      return ds->GetTypeName(colName);
-   }
+   if (ds && ds->HasColumn(colName))
+      colType = ds->GetTypeName(colName);
 
-   TBranch *branch = nullptr;
-   if (tree)
-      branch = tree->GetBranch(colName.c_str());
+   TBranch *branch = tree ? tree->GetBranch(colName.c_str()) : nullptr;
    if (branch) {
       // this must be a real TTree branch
       static const TClassRef tbranchelRef("TBranchElement");
       if (branch->InheritsFrom(tbranchelRef)) {
          // this branch is not a fundamental type, we can ask for the class name
-         return static_cast<TBranchElement *>(branch)->GetClassName();
+         std::string classname(static_cast<TBranchElement *>(branch)->GetClassName());
+         if (classname.compare(0, 7, "vector<") == 0) {
+            // column type is "vector<ValueType>", we read it as "TVec<ValueType>"
```

Comment:
Maybe here we could leverage the routines in TClassEdit to manipulate names. For example, we could consider:
- TClassEdit::IsSTLCont : in any case, all stl containers are stored the same and are accessible with TTreeReader, therewith TVec 
- TClassEdit::GetSplit: to get out the contained type name

The hard part was to write the code in this PR and surely this is a refinement...

Reply:
sure